### PR TITLE
Cleanup network usage in NBitcoin #6

### PR DESCRIPTION
**Pull request by fassadlr**
This mostly covers ConcurrentChain

### CODE REVIEW

**Comment by bokobza** [comment]
I'll do this one.

**Comment by fassadlr** [comment]
@bokobza I need to fix some tests ;)

**Comment by fassadlr** [comment]
@bokobza tests have been fixed :)

File: src/NBitcoin.Tests/ChainTests.cs [hunk 1]
```diff
@@ -404,13 +400,14 @@ public void CanForkSide()
...
         }
+
         [Fact]
         [Trait("UnitTest", "UnitTest")]
         public void CanForkSidePartialChain()
         {
             Block genesis = TestUtils.CreateFakeBlock();
-            var side = new ConcurrentChain(genesis.Header);
-            var main = new ConcurrentChain(genesis.Header);
+            var side = new ConcurrentChain(this.network);
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/NBitcoin.Tests/ChainTests.cs]: Fixed

File: src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs [hunk 2]
```diff
@@ -55,12 +55,11 @@ public void BuildTransactionNoSpendableTransactionsThrowsWalletException()
                         InternalAddresses = new List<HdAddress>()
                     });
 
-                var chain = new Mock<ConcurrentChain>();
                 BlockHeader blockHeader = this.network.Consensus.ConsensusFactory.CreateBlockHeader();
-                chain.Setup(c => c.Tip).Returns(new ChainedHeader(blockHeader, blockHeader.GetHash(), 1));
+                var chain = new ConcurrentChain(this.network, new ChainedHeader(blockHeader, blockHeader.GetHash(), 0));
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: Fixed

**Comment by bokobza** [comment]
Looks good, I'll wait for the tests.

**Pull request merged**